pilot/pkg/serviceregistry/kube/controller/ambient/helpers.go

```diff
@@ -65,21 +67,59 @@ func serviceToAddressInfo(s *workloadapi.Service) model.AddressInfo {
 	}
 }
 
-func byteIPToString(b []byte) string {
-	ip, _ := netip.AddrFromSlice(b)
+func mustByteIPToString(b []byte) string {
```

Comment:
Do we really need this? `byteIPToAddr().String()` works just as well, especially since this is just used in 2 places.